logic.py

```diff
@@ -101,7 +101,9 @@ def check_pw(email_address, pw):
     if not result:
         l('check_pw_bad_email', email=email_address)
         return None
-    if _mangle_pw(pw, result.pw) == result.pw:
+    salt = bytes.fromhex(result.pw[2:])
+    print(f"Password hash: {_mangle_pw(pw, result.pw)} and pw {salt}")
```

Comment:
f strings make this a 3.6 only port. If you ported it with code that runs cleanly under 2.7 and 3.4+ maybe upstream would take it as a PR.